packages/core/src/components/forms/_controls.scss

```diff
@@ -407,29 +424,28 @@ $control-indicator-spacing: $pt-grid-size !default;
     .#{$ns}-dark & {
       @include indicator-colors(
         "",
+        $dark-switch-text-color,
         $dark-switch-background-color,
         $dark-switch-background-color-hover,
         $dark-switch-background-color-active,
+        $dark-switch-text-color-disabled,
         $dark-switch-background-color-disabled,
         $dark-switch-indicator-background-color-disabled
       );
       @include indicator-colors(
         ":checked",
+        $dark-switch-checked-text-color,
         $dark-switch-checked-background-color,
         $dark-switch-checked-background-color-hover,
         $dark-switch-checked-background-color-active,
+        $dark-switch-checked-text-color-disabled,
         $dark-switch-checked-background-color-disabled,
         $dark-switch-indicator-background-color-disabled
       );
 
       .#{$ns}-control-indicator::before {
         background: $dark-switch-indicator-background-color;
-        box-shadow: $dark-button-box-shadow;
-      }
-
-      input:checked ~ .#{$ns}-control-indicator::before {
-        // inset shadow so it forms a dark line instead of blurring into the blue
```

Comment:
since the box shadow is different now, we don't get this blurring effect anymore without the inset